Foundation/NSNotificationQueue.swift

```diff
@@ -31,6 +33,16 @@ public class NSNotificationQueue : NSObject {
     internal let notificationCenter: NSNotificationCenter
     internal var asapList = NSNotificationList()
     internal var idleList = NSNotificationList()
+    internal lazy var idleRunloopObserver: CFRunLoopObserverRef = {
+        return CFRunLoopObserverCreateWithHandler(kCFAllocatorDefault, CUnsignedLong(1) << 5, true, 0) {[weak self] observer, activity in
```

Comment:
I am not certain if we want to mimic this behavior but the weak self is "incorrect" in comparison to the darwin version... however this seems to be much better behavior.

We should run valgrind to make certain the block here is correct since the layout is a bit in dispute for blocks